### ISSUE TITLE
Include html files while grepping?

### ISSUE THREAD

**Issue by ChillarAnand** [jorgenschaefer/elpy#574]
While working on django projects, if user searches for variable `foo` only *.py files are grepped. 

For this we can use projectile-grep but it does fuzzy match and it is very slower.

Also if we want to rename a variable in a view and use multi edit, it wont edit variables in template files.

May be we can include *.html files?

**Comment by jorgenschaefer** [comment]
Hm. Being able to adjust the `FILES` argument to `rgrep` on a per-project basis might be an idea. So basically replace `"*.py"` in `elpy-rgrep-symbol` with a defcustom? Would that solve your use case?

Renaming variables is tricky. Jedi/Rope won't look in Django templates, because parsing those is way beyond their scope. :-) You can use moccur I guess?

**Comment by ChillarAnand** [comment]
> Jedi/Rope won't look in Django templates

exactly. at some point of time elpy will support django templates?

> because parsing those is way beyond their scope

may be we have depend on another package like web-mode?

 [See this](https://github.com/jorgenschaefer/elpy/issues/427#issuecomment-97725148).

**Comment by jorgenschaefer** [comment]
> at some point of time elpy will support django templates?

As soon as someone adds support to Jedi/Rope.

> Even for this, we have depend on another package like web-mode?

I do not understand what you are trying to ask. Could you rephrase?

**Comment by ChillarAnand** [comment]
> I do not understand what you are trying to ask. Could you rephrase?

I think I misunderstood implementation. I thought we can support django templates by using web-mode(or similar package). As those packages can parse html files & can inform  about python contexts`({{}}, {%%})`, we can pass required information to those contexts using  jedi/rope.  

I wasn't aware that we need support from jedi/rope to implement this. 

Hope that makes  sense :smile:

**Comment by jorgenschaefer** [comment]
Ah, that explains my confusion :-)

Elpy does not do any parsing of Python itself. (Well, almost none) The code introspection happens entirely in Rope/Jedi. It's way too complex for Elpy. So features like these need to be supported on the Python side somehow.

**Comment by ChillarAnand** [comment]
should i post a feature request upstream?

**Comment by jorgenschaefer** [comment]
You can obviously post a feature request upstream, not sure why "we" would or what that would mean :-)

This is non-trivial, so I would not hold my breath!
